### ISSUE TITLE
100% CPU usage

### ISSUE THREAD

**Issue by GraveRaven** [securestate/king-phisher#156]
## Issue Description
When running king-phisher consumes 100% cpu all the time.

## Reproduction Steps
1. Start King Phisher

## Environment Details
Linux kali 4.6.0-kali1-amd64 #1 SMP Debian 4.6.4-1kali1 (2016-07-21) x86_64 GNU/Linux

## Error Details
As far as I can tell it could relate to a busy wait on the network interface.
Using strace I get the following with a huge amount of selects timing out.
```
<snip>
bind(5, {sa_family=AF_INET, sin_port=htons(80), sin_addr=inet_addr("0.0.0.0")}, 16) = 0
<snip>
select(6, [5], [], [], {0, 0})          = 0 (Timeout)
select(6, [5], [], [], {0, 0})          = 0 (Timeout)
select(6, [5], [], [], {0, 0})          = 0 (Timeout)
select(6, [5], [], [], {0, 0})          = 0 (Timeout)
<snip>
```

**Comment by zeroSteiner** [comment]
Thank you for this bug report, it's probably one of the best I've seen for King Phisher. I think I have an idea of what the problem is and should be able to test it later this week.

**Comment by zeroSteiner** [comment]
Alright so this was an issue with `select.select` being used to poll in [zeroSteiner/AdvancedHTTPServer](https://github.com/zeroSteiner/AdvancedHTTPServer) (which powers King Phisher). The problem was that select was being used to repeatedly poll the different subservers used for multiple port binds.

I fixed this in commit [6909dadd](https://github.com/zeroSteiner/AdvancedHTTPServer/commit/6909daddee2279e61937c8a5d6d1a981b1247d82) by using a wakeup file descriptor, removing the need to poll and check for shutdown requests. The fix is available in AdvancedHTTPServer v2.0.6 and can be applied to King Phisher immediately by using the command: `sudo python3 -m pip install --upgrade advancedhttpserver`. I'll push up a commit to the dev branch to use this patched version shortly.

**Comment by zeroSteiner** [comment]
Fixed and merged into master. Thanks again for the bug report. Go ahead and upgrade King Phisher and this should be resolved for you.
